flight/targets/flightd/fw/main.c

```diff
@@ -106,14 +106,14 @@ void initTask(void *parameters)
 
 	PIOS_SYS_Init();
 
-	/* board driver init */
-	PIOS_Board_Init();
+	/* Initialize the task monitor library */
+	TaskMonitorInitialize();
```

Comment:
I don't really like this going before Board_Init, and we need to make sure that _never_ happens on embedded hardware. Maybe we should just rename PIOS_Board_Init for flightd to something else to avoid any confusion.

Reply:
THis goes (close to) first thing in pios_board_init.  And should get factored out everywhere IMO.

Board init starts tasks that expect to be able to use the task monitor subsystem.

Reply:
I was looking into doing this on embedded and it looks like we already do ;)

https://github.com/d-ronin/dRonin/blob/next/flight/PiOS/STM32/chibi_main.c#L132

Reply:
Ah. I guess UAVOs don't really depend too much on PIOS_Board_Init as long as they're not saved to flash, it's encouraging someone to do that one day though. :stuck_out_tongue: